api/model/src/testUriCompliance/java/org/projectnessie/model/TestElementsEncoding.java

```diff
@@ -495,4 +509,38 @@ static List<List<String>> randomRoundTrips() {
     }
     return r;
   }
+
+  @ParameterizedTest
+  @MethodSource
+  void allCharacters(String element) {
+    List<String> elements = List.of(element);
+    String escaped = Util.toPathStringEscaped(elements);
+    String canonical = Util.toCanonicalString(elements);
+
+    String uriEncodedEscaped = URLEncoder.encode(escaped, UTF_8);
+    HttpURI uri = HttpURI.from(format("http://hostname/foo/%s", uriEncodedEscaped));
+    soft.assertThat(uri.getViolations())
+        .describedAs("escaped: %s -  url: %s", escaped, uriEncodedEscaped)
+        .isEmpty();
+
+    List<String> elementsFromEscaped = Util.fromPathString(escaped);
+    List<String> elementsFromCanonical = Util.fromPathString(canonical);
+
+    soft.assertThat(elementsFromEscaped).containsExactlyElementsOf(elements);
+    soft.assertThat(elementsFromCanonical).containsExactlyElementsOf(elements);
+  }
+
+  static Stream<String> allCharacters() {
+    int split = 32;
+    return IntStream.rangeClosed(32 / split, 0xffff / split)
+        .mapToObj(
+            i -> {
+              char c = (char) (i * split);
+              StringBuilder sb = new StringBuilder(split);
+              for (int i1 = 0; i1 < 8; i1++) {
```

Comment:
Each iteration span includes 32 chars, but we only return the first 8?

Reply:
Argh - left-over.

However, 0x7f triggered a test failure. Just decided to forbid `DEL`

Reply:
Forbidding `DEL` SGTM